documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,347 @@
+
+== Configuring the Cassandra database
+The main two properties to configure are `contact-points`, to access the Cassandra database, and `local-datacenter`, which is required by the driver.
+
+A sample configuration should look like this:
+
+[source,properties]
+----
+# configure the Cassandra client for a replica set of two nodes
+quarkus.cassandra.contact-points={cassandra_ip}:9042
+quarkus.cassandra.load-balancing-policy.local-datacenter={dc_name}
+----
+
+In this example, we are using a single instance running on localhost:
+
+[source,properties]
+----
+# configure the Cassandra client for a single instance on localhost
+quarkus.cassandra.contact-points=127.0.0.1:9042
+quarkus.cassandra.load-balancing-policy.local-datacenter=datacenter1
+----
+
+You can configure all Java driver settings using `application.conf` or `application.json` files.
+They need to be located in the resources directory of your application.
+They will be passed automatically to the underlying driver configuration mechanism.
+Settings defined in `application.properties` with the `quarkus.cassandra` prefix will have priority over settings defined in `application.conf` or `application.json`.
+
+To see the full list of settings, please refer to the {reference-conf-url}[reference.conf].
+
+== Running a Cassandra Database
+By default, `CassandraClient` is configured to access a local Cassandra database on port 9042 (the default Cassandra port).
+If you have a local running database on this port, you should assert that:
+`quarkus.cassandra.load-balancing-policy.local-datacenter` setting is equal to dc of your local cassandra instance.
+This value can be found with `SELECT data_center FROM system.local` query.
+
+If you want to use Docker to run a Cassandra database, you can use the following command to launch one:
+[source,shell]
+----
+docker run \
+   --name local-cassandra-instance \
+   -p 7000:7000 \
+   -p 7001:7001 \
+   -p 7199:7199 \
+   -p 9042:9042 \
+   -p 9160:9160 \
+   -p 9404:9404 \
+   -d \
+   launcher.gcr.io/google/cassandra3
+----
+Note that only the 9042 port is required. All others all optional but provide enhanced features like JMX monitoring of the C* instance.
+
+Next you need to create the keyspace and table that will be used by your application:
+[source,shell]
+----
+docker exec -it local-cassandra-instance cqlsh -e "CREATE KEYSPACE IF NOT EXISTS k1 WITH replication = {'class':'SimpleStrategy', 'replication_factor':1}"
+docker exec -it local-cassandra-instance cqlsh -e "CREATE TABLE IF NOT EXISTS k1.fruit(id text, name text, description text, PRIMARY KEY((id), name))"
+----
+
+If you're running Cassandra locally you can execute the cqlsh commands directly:
+[source,shell]
+----
+cqlsh -e "CREATE KEYSPACE IF NOT EXISTS k1 WITH replication = {'class':'SimpleStrategy', 'replication_factor':1}
+cqlsh -e "CREATE TABLE IF NOT EXISTS k1.fruit(id text, name text, description text, PRIMARY KEY((id), name))
+----
```

Comment:
Here we _can_ point people to ```cqlsh -f setup_db.cql```, right?  Might be simpler than having to show the statements directly inline again.

Reply:
I disagree. It is simpler to have those statements in the docs. Otherwise, the user will need to lookup another file.

Reply:
I"m not sure I agree, but I don't feel terribly strongly about it... so let's keep them as they are.